cmrtlib/agnostic/share/cm_queue_base.h

```diff
@@ -59,6 +59,7 @@ const CM_QUEUE_CREATE_OPTION CM_DEFAULT_QUEUE_CREATE_OPTION = { CM_QUEUE_TYPE_RE
 class CmQueue
 {
 public:
+    virtual ~CmQueue() = default;
```

Comment:
CmQueue and CmDevice are CM APIs which can be called directly by applications. We can't just insert a destructor here for backward compatibility. We will check whether it can maintain the backward compatibility if we add the destructor in protected domain.

Reply:
Ok, so you suggest to move destructor to protected: domain of class?

Reply:
Yes. Directly deleting CmQueue and CmDevice class is forbidden, so moving the destructor in protected: domain can avoid directly deleting these two classes, and can avoid this kind of warning.
We are running a sanity test with your changes + protected destructors. Once it passes, we can merge this change.

Reply:
about the protected destructor, you can refer to b9c82263251566ca129566c40c7a934fe5904162

Reply:
Thanks a lot